app/models/User.coffee

```diff
@@ -429,7 +429,11 @@ module.exports = class User extends CocoModel
     stripe.planID = 'basic'
     stripe.token = token.id
     @set({stripe})
-    return me.patch({headers: {'X-Change-Plan': 'true'}})
+    return me.patch({headers: {'X-Change-Plan': 'true'}}).then =>
+      unless utils.isValidEmail(@get('email'))
+        @set({email: token.email})
+        me.patch()
+      return Promise.resolve()
```

Comment:
This alters what the Promise returns. Looks like this won't have any effect, and will be replaced with API, so no change needed, but something to be aware of.